src/components/SwitchToggle.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import styled from 'styled-components/macro';
+import PropTypes from 'prop-types';
+
+const Checkbox = styled.input.attrs({ type: 'checkbox' })`
+  border: 0;
+  clip: rect(0 0 0 0);
+  height: 1px;
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  width: 1px;
+`;
+
+const Label = styled.label`
+  align-items: center;
+  display: flex;
+  min-height: 2rem;
+  padding-left: 3.5rem;
+  position: relative;
+  user-select: none;
+  &::before {
+    background-color: #fff;
+    border: 1px solid
+      ${({ checked, theme }) => (checked ? theme.button : '#ece9f1')};
+    border-radius: 100rem;
+    box-shadow: ${({ theme }) => theme.boxShadowSmall};
+    content: '';
+    display: block;
+    height: 1.75rem;
+    left: 0;
+    position: absolute;
+    top: 0.125rem;
+    transition: border-color ${({ theme }) => theme.transitionEase};
+    width: 3rem;
+  }
+  &::after {
+    background-color: ${({ checked, theme }) =>
+      checked ? theme.button : '#ece9f1'};
+    border-radius: 50%;
+    content: '';
+    height: 1.5rem;
+    left: 0.125rem;
+    position: absolute;
+    top: 0.25rem;
+    transform: translateX(${props => (props.checked ? '1.25rem' : 0)});
+    transition: background-color ${({ theme }) => theme.transitionEase},
+      transform ${({ theme }) => theme.transitionEase};
+    width: 1.5rem;
+    transition: all 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275),
+      padding 0.3s ease, margin 0.3s ease;
```

Comment:
There is another `transition` above, this one is unnecessary. If you want to update the timing-function, you should probably add a new one in the `theme.js`, like `transitionEaseOut: 'cubic-bezier(0.175, 0.885, 0.3, 1.275)', and use it here on line 48-49. Also, you should never use `all` in transitions (just the ones changing).